### PR TITLE
[33392] Cache allowed requests of schema to prevent double loading assignees/responsibles

### PR DESCRIPTION
**Pull request by oliverguenther**
:warning: Includes https://github.com/opf/openproject/pull/8393

https://community.openproject.com/wp/33392

### CODE REVIEW

File: frontend/src/app/components/wp-buttons/wp-status-button/wp-status-button.component.ts [hunk 1]
```diff
@@ -35,6 +35,8 @@ import {Highlighting} from "core-components/wp-fast-table/builders/highlighting/
 import {HalResource} from "core-app/modules/hal/resources/hal-resource";
 import {WorkPackageCacheService} from "core-components/work-packages/work-package-cache.service";
 import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
+import {concatAll} from "rxjs/operators";
+import {combineLatest} from "rxjs";
```
Comment (openprojectci) [inline frontend/src/app/components/wp-buttons/wp-status-button/wp-status-button.component.ts:40]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/64ce82c1eb6f239ac1705728426be82eb4fe79c6/frontend/src/app/components/wp-buttons/wp-status-button/wp-status-button.component.ts#L40"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts [hunk 2]
```diff
@@ -43,6 +43,7 @@ import {IFieldSchema} from "core-app/modules/fields/field.base";
 import {EditFieldHandler} from "core-app/modules/fields/edit/editing-portal/edit-field-handler";
 import {EditingPortalService} from "core-app/modules/fields/edit/editing-portal/editing-portal-service";
 import {EditFormRoutingService} from "core-app/modules/fields/edit/edit-form/edit-form-routing.service";
+import {ResourceChangesetCommit} from "core-app/modules/fields/edit/services/hal-resource-editing.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts:47]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/64ce82c1eb6f239ac1705728426be82eb4fe79c6/frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L47"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/4c575217bb2ce7184334bc51553c7701deb5da0c/frontend/src/app/modules/fields/edit/field-types/select-edit-field.component.ts#L45">frontend/src/app/modules/fields/edit/field-types/select-edit-field.component.ts#L45</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/4c575217bb2ce7184334bc51553c7701deb5da0c/frontend/src/app/modules/fields/edit/field-types/te-work-package-edit-field.component.ts#L41">frontend/src/app/modules/fields/edit/field-types/te-work-package-edit-field.component.ts#L41</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts [hunk 3]
```diff
@@ -11,27 +11,18 @@ import {WorkPackageViewFiltersService} from "core-app/modules/work_packages/rout
...
 import {UrlParamsHelperService} from "core-components/wp-query/url-params-helper";
 import {StateService} from "@uirouter/core";
-import {DebouncedEventEmitter} from "core-components/angular/debounced-event-emitter";
-import {skip} from "rxjs/internal/operators";
-import {ApiV3Filter} from "core-components/api/api-v3/api-v3-filter-builder";
+import {debounceTime, skip, take} from "rxjs/internal/operators";
 import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
-import {componentDestroyed} from "@w11k/ngx-componentdestroyed";
+import {Observable} from "rxjs";
+import {BoardFiltersService} from "core-app/modules/boards/board/board-filter/board-filters.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts:18]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/20825e334564e0fd5335f3e189a9eb5493b01228/frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts#L18"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/boards/boards-root/boards-root.component.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ import {BoardAssigneeActionService} from "core-app/modules/boards/board/board-ac
```
Comment (openprojectci) [inline frontend/src/app/modules/boards/boards-root/boards-root.component.ts:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/20825e334564e0fd5335f3e189a9eb5493b01228/frontend/src/app/modules/boards/boards-root/boards-root.component.ts#L8"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Review by HDinger — APPROVED** [review]

**Pull request merged**